vendor/blockchain_contracts/src/bitcoin/rfc003/bitcoin_htlc.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{
+    fit_into_placeholder_slice::{BitcoinTimestamp, FitIntoPlaceholderSlice},
+    SecretHash,
+};
+use bitcoin_support::{Address, Hash160, Network, Script};
+use bitcoin_witness::{UnlockParameters, Witness, SEQUENCE_ALLOW_NTIMELOCK_NO_RBF};
+use hex_literal::hex;
+use secp256k1_support::KeyPair;
+
+// contract template RFC: https://github.com/comit-network/RFCs/blob/master/RFC-005-SWAP-Basic-Bitcoin.md#contract
+pub const CONTRACT_TEMPLATE: [u8;97] = hex!("6382012088a82010000000000000000000000000000000000000000000000000000000000000018876a9143000000000000000000000000000000000000003670420000002b17576a91440000000000000000000000000000000000000046888ac");
+
+#[derive(Debug)]
+pub enum UnlockingError {
+    WrongSecret {
+        got: SecretHash,
+        expected: SecretHash,
+    },
+    WrongPubkeyHash {
+        got: [u8; 20],
+        expected: [u8; 20],
+    },
+}
+
+#[derive(Debug, Clone)]
+pub struct BitcoinHtlc {
+    script: Vec<u8>,
+    expiry: u32,
+}
+
+impl BitcoinHtlc {
+    pub fn new(
+        expiry: u32,
+        refund_identity: Hash160,
+        redeem_identity: Hash160,
+        secret_hash: [u8; 32],
+    ) -> Self {
+        let mut contract = CONTRACT_TEMPLATE.to_vec();
+        SecretHash(secret_hash).fit_into_placeholder_slice(&mut contract[7..39]);
+        redeem_identity.fit_into_placeholder_slice(&mut contract[43..63]);
+        BitcoinTimestamp(expiry).fit_into_placeholder_slice(&mut contract[65..69]);
+        refund_identity.fit_into_placeholder_slice(&mut contract[74..94]);
+
+        BitcoinHtlc {
+            script: contract,
+            expiry,
+        }
+    }
+
+    pub fn compute_address(&self, network: Network) -> Address {
+        Address::p2wsh(&self.to_script(), network.into())
+    }
+
+    pub fn unlock_with_secret(&self, keypair: KeyPair, secret: [u8; 32]) -> UnlockParameters {
```

Comment:
🔧 didn't you mention it'd be good to have a rfc003 trait that defines these functions common to all rfc003 contracts? (`unlock_with_secret`, `unlock_after_timeout`)

Reply:
Yeah but that would be quite a refactoring :o 

I'll create a ticket since it is completely unrelated to the current PR.

Reply:
Done: https://github.com/comit-network/comit-rs/issues/1085